systemvm/patches/debian/config/opt/cloud/bin/cs_dhcp.py

```diff
@@ -34,16 +34,14 @@ def merge(dbag, data):
     return dbag
 
 
-def search(dbag, name):
+def search(dbag, name, gateway):
     """
     Dirty hack because CS does not deprovision hosts
     """
     hosts = []
     for o in dbag:
         if o == 'id':
             continue
-        print "%s %s" % (dbag[o]['host_name'], name)
-        if dbag[o]['host_name'] == name:
+        print "%s %s %s" % (dbag[o]['host_name'], name, gateway)
+        if dbag[o]['host_name'] == name and dbag[o]['default_gateway'] == gateway :
             hosts.append(o)
-    for o in hosts:
```

Comment:
@ustcweizhou @niteshsarda on searching should we be removing the hosts from databag? i.e. should be keep or remove:
```
for o in hosts:
    del(dbag[o])
```